File: python-legacy-build/magic/magic_on_npc.py
from magic_settings import SET_INS

class OnNPC:
    def kww(self, dmg):
        if SET_INS.gloves == 'Enhanced Kerapac':
            return int(dmg + (dmg * 0.4 * SET_INS.kww))
        else:
            return int(dmg + (dmg * 0.25 * SET_INS.kww))
        
    def vuln(self, dmg):
        return int(dmg + (dmg * 0.1 * SET_INS.vuln))
    
    def sc(self,dmg):
        return int(dmg + (dmg * 0.15 * SET_INS.sc))
    
    def cryptbloom(self, dmg):
        return int(dmg + (dmg * 0.1 * SET_INS.cryptbloom))
    
    def slayerperk(self, dmg):
        return int(dmg + (dmg * 0.07 * SET_INS.slayerPerk))
    
    def slayersigil(self, dmg):
        return int(dmg + (dmg * 0.15 * SET_INS.slayerSigil))
    
    def aura(self, dmg):
        boost = 0
        for b in SET_INS.utils.boosts:
            if b['name'] == SET_INS.aura:
                boost = b['magic_dmg_percent']
        
        return int(dmg + int(dmg * boost))
    
    def meta(self, dmg):
        return int(dmg * 1.625)
    
    def scrimshaw(self, dmg):
        if SET_INS.pocket == 'Superior scrimshaw of elements':
            return int(dmg * 1.0666)
        elif SET_INS.pocket == 'Scrimshaw of elements':
            return int(dmg * 1.05)
        else:
            return dmg
        
    def computeOnNpc(self, reg, sun):
        reg = self.kww(reg)
        sun = self.kww(sun)
        reg = self.vuln(reg)
        sun = self.vuln(sun)
        reg = self.cryptbloom(reg)
        sun = self.cryptbloom(sun)
        reg = self.slayerperk(reg)
        sun = self.slayerperk(sun)
        reg = self.slayersigil(reg)
        sun = self.slayersigil(sun)
        reg = self.aura(reg)
        meta = self.meta(reg)
        reg = self.scrimshaw(reg)
        meta = self.scrimshaw(meta)
        sun = self.scrimshaw(sun)
        
        return [reg, sun, meta]
        
    def computeOnNpcCrit(self, reg, sun):
        reg = self.kww(reg)
        sun = self.kww(sun)
        reg = self.vuln(reg)
        sun = self.vuln(sun)
        reg = self.sc(reg)
        sun = self.sc(sun)
        reg = self.cryptbloom(reg)
        sun = self.cryptbloom(sun)
        reg = self.slayerperk(reg)
        sun = self.slayerperk(sun)
        reg = self.slayersigil(reg)
        sun = self.slayersigil(sun)
        reg = self.aura(reg)
        meta = self.meta(reg)
        reg = self.scrimshaw(reg)
        meta = self.scrimshaw(meta)
        sun = self.scrimshaw(sun)
        
        return [reg, sun, meta]
        
    
    

File: magic/magic_on_npc.js
class OnNPC {
    kww(dmg, kww, ench) {
      if (ench === true) {
        return Math.floor(dmg + (dmg * 0.4 * kww));
      } else {
        return Math.floor(dmg + (dmg * 0.25 * kww));
      }
    }
  
    vuln(dmg, flag) {
      return Math.floor(dmg + (dmg * 0.1 * flag));
    }
  
    sc(dmg, flag) {
      return Math.floor(dmg + (dmg * 0.15 * flag));
    }
  
    cryptbloom(dmg, flag) {
      return Math.floor(dmg + (dmg * 0.1 * flag));
    }
  
    slayerperk(dmg, flag) {
      return Math.floor(dmg + (dmg * 0.07 * flag));
    }
  
    slayersigil(dmg, flag) {
      return Math.floor(dmg + (dmg * 0.15 * flag));
    }
  
    aura(dmg, boost) {
      return Math.floor(dmg + (dmg * boost));
    }
  
    meta(dmg) {
      return Math.floor(dmg * 1.625);
    }
  
    scrimshaw(dmg, scrimshaw) {
      if (scrimshaw === 'superior') {
        return Math.floor(dmg * 1.0666);
      } else if (scrimshaw === 'base') {
        return Math.floor(dmg * 1.05);
      } else {
        return dmg;
      }
    }
  
    calcOnNpc(reg, sun, kww, ench, vuln, cryptbloom, slayerperk, slayersigil, aura, scrimshaw, superior) {
      reg = this.kww(reg, kww, ench);
      sun = this.kww(sun, kww, ench);
      reg = this.vuln(reg, vuln);
      sun = this.vuln(sun, vuln);
      reg = this.cryptbloom(reg, cryptbloom);
      sun = this.cryptbloom(sun, cryptbloom);
      reg = this.slayerperk(reg, slayerperk);
      sun = this.slayerperk(sun, slayerperk);
      reg = this.slayersigil(reg, slayersigil);
      sun = this.slayersigil(sun, slayersigil);
      reg = this.aura(reg, aura);
      let meta = this.meta(reg);
      reg = this.scrimshaw(reg, scrimshaw, superior);
      meta = this.scrimshaw(meta, scrimshaw, superior);
      sun = this.scrimshaw(sun, scrimshaw, superior);
  
      return [reg, sun, meta];
    }
  
    calcOnNpcCrit(reg, sun, kww, ench, vuln, sc, cryptbloom, slayerperk, slayersigil, aura, scrimshaw, superior) {
      reg = this.kww(reg, kww, ench);
      sun = this.kww(sun, kww, ench);
      reg = this.vuln(reg, vuln);
      sun = this.vuln(sun, vuln);
      reg = this.sc(reg, sc);
      sun = this.sc(sun, sc);
      reg = this.cryptbloom(reg, cryptbloom);
      sun = this.cryptbloom(sun, cryptbloom);
      reg = this.slayerperk(reg, slayerperk);
      sun = this.slayerperk(sun, slayerperk);
      reg = this.slayersigil(reg, slayersigil);
      sun = this.slayersigil(sun, slayersigil);
      reg = this.aura(reg, aura);
      let meta = this.meta(reg);
      reg = this.scrimshaw(reg, scrimshaw, superior);
      meta = this.scrimshaw(meta, scrimshaw, superior);
      sun = this.scrimshaw(sun, scrimshaw, superior);
  
      return [reg, sun, meta];
    }
  }
  
module.exports = OnNPC;